### PR TITLE
Fix reborrow verifier to look through borrowed from

### PR DESCRIPTION
**Pull request by meg-gupta**
rdar://141490868

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by nate-chandler — APPROVED** [review]
LGTM (a question about `build-presets.ini`)

File: utils/build-presets.ini [hunk 1]
```diff
@@ -1453,6 +1453,7 @@ skip-build-tvos
 skip-test-tvos
 skip-build-watchos
 skip-test-watchos
+skip-build-xros
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply utils/build-presets.ini:1456]: oops no. I'll delete it.

**Comment by meg-gupta** [comment]
@swift-ci test

**Pull request merged**